protocol.md

```diff
@@ -77,23 +89,27 @@ Functions:
 Out of band:
 
 -   Agree on a PAYLOAD_TYPE and cryptographic details, optionally including
-    KEYID.
+    KEYID and EXTENSION.
 
 To sign:
 
 -   Serialize the message according to PAYLOAD_TYPE. Call the result
     SERIALIZED_BODY.
 -   Sign PAE(UTF8(PAYLOAD_TYPE), SERIALIZED_BODY). Call the result SIGNATURE.
 -   Optionally, compute a KEYID.
+-   Optionally, include signature specific information as an EXTENSION.
 -   Encode and transmit SERIALIZED_BODY, PAYLOAD_TYPE, SIGNATURE, and KEYID,
     preferably using the recommended [JSON envelope](envelope.md).
 
 To verify:
 
--   Receive and decode SERIALIZED_BODY, PAYLOAD_TYPE, SIGNATURE, and KEYID, such
-    as from the recommended [JSON envelope](envelope.md). Reject if decoding
-    fails.
+-   Receive and decode SERIALIZED_BODY, PAYLOAD_TYPE, SIGNATURE, KEYID, and
+    EXTENSION such as from the recommended [JSON envelope](envelope.md). Reject
+    if decoding fails.
 -   Optionally, filter acceptable public keys by KEYID.
+-   Optionally, use EXTENSION fields if available to obtain information that may
```

Comment:
It isn't optional to use the EXTENSION fields, right?  It's just that one may not understand them because they are optional.  In other words, I believe a "correct" client must verify all extension fields it understands.

Reply:
I think you're right, at least in in-toto/TUF use cases. How should we frame this?

Reply:
I took a pass at editing this. Please take a look.

Reply:
How does this read? I'll make the word "may" all caps, but I think it handles @MarkLodato's comment about an extension that may not actually be relevant.

```
-   If EXTENSION is set, use its fields to obtain information that may be
    required for verification. Reject any significant fields that the verifier
    must establish separately, such as the public key for a signature or a root
    certificate for an X.509 certificate chain.
```

Reply:
> @JustinCappos I don't see how this follows. Suppose I am expecting a DSSE using a pre-shared key. Someone adds an invalid "Sigstore" extension to the DSSE but doesn't touch any other field, so the DSSE still passes if you ignore the extension. Why would a client be REQUIRED to reject such a DSSE? I can see that they MAY do that, but it doesn't seem mandatory to me.

Would a client that checks the flawed Sigstore extension reject it, even if other things pass?  It just seems odd to me that you could remove or not understand an extension as a way to have something pass.

Reply:
Oh, @trishankatdatadog just pointed out that the text already says this:

> Additionally, extensions MUST NOT contain any information such that the
> signature verification fails in its presence and passes in its absence.

It sounds like you are disagreeing with this statement, @JustinCappos?